proxy-node-shared/src/main/java/uk/gov/ida/notification/configuration/CloudHsmCredentialConfiguration.java

```diff
@@ -37,9 +38,17 @@ public CloudHsmCredentialConfiguration(
             JCEMapper.setProviderId("Cavium");
             KeyStore cloudHsmStore = KeyStore.getInstance("Cavium");
             cloudHsmStore.load(null, null);
+
+            final Key key = cloudHsmStore.getKey(hsmKeyLabel, null);
             BasicX509Credential credential = new BasicX509Credential(
                     certificate,
-                    (PrivateKey) cloudHsmStore.getKey(hsmKeyLabel, null));
+                    (PrivateKey) key);
+
+            this.keyHandle = (Long) ClassLoader.getSystemClassLoader()
+                                                     .loadClass("com.cavium.key.CaviumKey")
+                                                     .getMethod("getHandle")
+                                                     .invoke(key);
```

Comment:
Indentation is a bit off here - in the future, can we please use the IntelliJ formatting to make it consistent